app/pages/UcsfClientEligibilityPage/UcsfClientEligibilityPage.tsx

```diff
@@ -7,108 +7,95 @@ import { Section } from 'components/ucsf/Section/Section';
 import { Layout } from 'components/ucsf/Layout/Layout';
 
 import {
-  eligibilityData,
-  UcsfEligibilityDictionary,
+  eligibilityMap,
   Eligibility,
   EligibilityGroup,
-} from './ucsfEligibilities';
+} from './ucsfEligibilitiesMap';
 import styles from './UcsfClientEligibilityPage.module.scss';
 
-const ClientEligibilities = ({ rawEligibilityData, resourceSlug }: {
-  rawEligibilityData: UcsfEligibilityDictionary;
+interface SelectedEligibilities {
+  [key: string]: boolean;
+}
+
+const ClientEligibilities = ({
+  resourceEligibilityGroups, selectedEligibilities, setSelectedEligibilities, resourceSlug,
+}: {
+  resourceEligibilityGroups: EligibilityGroup[];
+  selectedEligibilities: SelectedEligibilities;
+  setSelectedEligibilities: (categories: SelectedEligibilities) => void;
   resourceSlug: string;
 }) => {
   useEffect(() => {
-    setEligibilityGroupList(rawEligibilityData[resourceSlug]);
+    setEligibilityGroupList(resourceEligibilityGroups);
   }, [resourceSlug]);
 
   const [eligibilityGroupList, setEligibilityGroupList] = useState<EligibilityGroup[]>(
-    rawEligibilityData[resourceSlug],
+    resourceEligibilityGroups,
   );
 
-  // Todo: This setEligibilityGroup and toggleChecked logic could change pretty drastically
-  // once the API returns eligibility data. The shape of that data is still under discussion
-  // between product and dev
-  const setEligibilityGroup = (index: number, updatedEligibilityGroup: EligibilityGroup) => {
-    const updatedList = [
-      ...eligibilityGroupList.slice(0, index),
-      updatedEligibilityGroup,
-      ...eligibilityGroupList.slice(index + 1),
-    ];
-
-    setEligibilityGroupList(updatedList);
-  };
-
-  const toggleChecked = (eligibilityGroup: EligibilityGroup, eligibilityIndex: number) => {
-    const targetEligibility = eligibilityGroup.eligibilities[eligibilityIndex];
-    const targetToggleState = !targetEligibility.checked;
-    let updatedEligibilities;
-
-    // If a user toggles "See all", we (un)check all of the eligibilities accordingly
-    if (targetEligibility.name === 'See all') {
-      updatedEligibilities = massToggleGroupEligibilities(
-        eligibilityGroup.eligibilities,
-        targetToggleState,
-      );
+  const handleEligibilityClick = (
+    eligibility: Eligibility,
+    eligibilities: Eligibility[],
+  ) => {
+    const seeAllEligibilityItem = eligibilities.find(e => e.isSeeAll);
+    const eligibilityCheckedId = eligibility.checkedId;
```

Comment:
Nit: Is this variable providing any value over just writing out `eligibility.checkedId` in the places where it is used? The variable's name is basically the exact same thing, so it's neither more expressive nor more concise.